src/self/subResourceWrapper/deploy.bicep

```
@@ -271,4 +365,67 @@ module createLzRoleAssignmentsRsgsNotSelf '../../carml/v0.6.0/Microsoft.Authoriz
   }
 }]
 
+module createResourceGroupForDeploymentScript '../../carml/v0.6.0/Microsoft.Resources/resourceGroups/deploy.bicep' = if (!empty(resourceProviders) || !empty(resourceProvidersFeatures)) {
+  scope: subscription(subscriptionId)
+  name: deploymentNames.createResourceGroupForDeploymentScript
+  params: {
+    name: deploymentScriptResourceGroupName
+    location: deploymentScriptLocation
+    lock: ''
+    enableDefaultTelemetry: enableTelemetryForCarml
+  }
+}
+
+module createDeploymentScriptManagedIdentity '../../carml/v0.6.0/Microsoft.ManagedIdentity/userAssignedIdentity/deploy.bicep' = if (!empty(resourceProviders) || !empty(resourceProvidersFeatures)) {
+  scope: resourceGroup(subscriptionId, deploymentScriptResourceGroupName)
+  dependsOn: [
+    createResourceGroupForDeploymentScript
+  ]
+  name: deploymentNames.createDeploymentScriptManagedIdentity
+  params: {
+    location: deploymentScriptLocation
+    name: deploymentScriptManagedIdentityName
+    enableDefaultTelemetry: enableTelemetryForCarml
+  }
+}
+
+module createRoleAssignmentsDeploymentScript '../../carml/v0.6.0/Microsoft.Authorization/roleAssignments/deploy.bicep' = if (!empty(resourceProviders) || !empty(resourceProvidersFeatures)) {
+  dependsOn: [
+    createDeploymentScriptManagedIdentity
+  ]
+  name: take('${deploymentNames.createRoleAssignmentsDeploymentScript}', 64)
+  params: {
+    location: deploymentScriptLocation
+    principalId: createDeploymentScriptManagedIdentity.outputs.principalId
+    roleDefinitionIdOrName: 'Contributor'
+    subscriptionId: subscriptionId
+    enableDefaultTelemetry: enableTelemetryForCarml
+  }
+}
+
+module registerResourceProviders '../../carml/v0.6.0/Microsoft.Resources/deploymentScripts/deploy.bicep' = if (!empty(resourceProviders) || !empty(resourceProvidersFeatures)) {
+  scope: resourceGroup(subscriptionId, deploymentScriptResourceGroupName)
+  name: deploymentNames.registerResourceProviders
+  params: {
+    name: deploymentScriptName
+    kind: 'AzurePowerShell'
+    azPowerShellVersion: '3.0'
+    cleanupPreference: 'Always'
+    enableDefaultTelemetry: enableTelemetryForCarml
+    location: deploymentScriptLocation
+    retentionInterval: 'P1D'
+    timeout: 'PT1H'
+    runOnce: true
+    userAssignedIdentities: {
+      '${createDeploymentScriptManagedIdentity.outputs.resourceId}': {}
+    }
+    arguments: '-resourceProviders \'${resourceProvidersFormatted}\' -resourceProvidersFeatures \'${resourceProvidersFeaturesFormatted}\' -subscriptionId ${subscriptionId}'
+    scriptContent: loadTextContent('../../../.github/scripts/Register-SubResourceProviders.ps1')
+  }
+}
+
 // OUTPUTS
+
+output failedProviders string = !empty(resourceProviders) || !empty(resourceProvidersFeatures) ? registerResourceProviders.outputs.outputs['failedProvidersRegistrations'] : ''
```

```suggestion
output failedProviders string = !empty(resourceProviders) ? registerResourceProviders.outputs.outputs['failedProvidersRegistrations'] : ''
```

Can be simplified right as no OR needed?